docs/rfc/0003_vex_support.md

```diff
@@ -0,0 +1,218 @@
+
+The use of this format can sensitively reduce the number of vulnerabilities 
+in the final vulnerability report, which more often is full of false positive
+entries.
+
+In order to reduce the noise, VEX is the right choise if you are dealing with
+SBOMs and OCI images.
+
+That said, we want to add support for:
+
+* keeping configuration of multiple VexHub repositories
+
+* support private VexHub repos with credentials and auth tokens
+
+* use VEX docuements to scan images
+
+## Examples / User Stories
+
+[examples]: #examples
+
+<!---
+Examples of how the feature will be used. Interactions should show the action
+and the response. When appropriate, provide user stories in the form of "As a
+[role], I want [feature], so [that]."
+--->
+
+### User story #1
+
+As a user, I want to scan registries on my infrastracture filtering out the 
+number of CVEs detected by SBOMbastic, removing false positives as much as possible.
+
+### User story #2
+
+As a user, I want to configure the registries scan with appropriate VEX files 
+depending on the content of the registry (eg. test, staging or prod images), 
+so that the vulnerability report will have an accurate result.
+
+# Detailed design
+
+[design]: #detailed-design
+
+<!---
+This is the bulk of the RFC. Explain the design in enough detail for somebody
+familiar with the product to understand, and for somebody familiar with the
+internals to implement.
+
+This section should cover architecture aspects and the rationale behind
+disruptive technical decisions (when applicable), as well as corner-cases and
+warnings.
+--->
+
+This RFC introduces support for VEX files when scanning registries.
+From now on, the Registry CRD will have an additional field to refer the VexHub
+resources we want to use to scan it.
+
+Suppose you are scanning the registry of a dedicated department in your company.
+This registry will host specific kind of images depending on the scope of the 
+department. In order to make the scan as much accurate as possible, we are 
+going to provide a new clusterwide CRD called VexHub. This CRD will hold 
+configuration of the VexHub repositories, so that the Registry CRD will refer 
+to these VexHub resources.
+
+Since VexHub CRD is clusterwide, this means that you can use the same 
+configuration across multiple registries.
+
+## VexHub CRD
+
+To configure a new VexHub, the user will need to apply a new manifest as follow:
+
+```yaml
+apiVersion: sbombastic.rancher.io/v1alpha1
+kind: VexHub
+metadata:
+  name: vendor-vexhub
+  namespace: default
+spec:
+  url: https://vexhub.vendor.com
+  credentials:
+    secretRef:
+      name: vendor-vexhub-secret
+      key: access-token
+```
+
+As you can see the VexHub CRD has a `secretRef` field, which means that you 
+need to provide a Secret to configure the credentials (if any):
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: vexhub-auth
+  namespace: security-team
+type: Opaque
+data:
+  token: <base64-encoded-token>
```

Comment:
or username and password: https://trivy.dev/latest/docs/supply-chain/vex/repo/#authentication